File: 20-valid-parentheses/20-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        unordered_map<char, char> mp = { {')', '('}, {']', '['}, {'}', '{'} };
        for (int i = 0; i < s.size(); i ++){
            if (s[i] == '(' || s[i] == '[' || s[i] == '{')
                st.push(s[i]);
            else{
                if (st.size() && mp[s[i]] == st.top()) st.pop();
                else return false;
            }
        }
        return st.empty();
    }
};

File: 20-valid-parentheses/20-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
                Stack<Character> stack = new Stack<>();
        for(Character ch : s.toCharArray()){
            boolean condition = ch.equals('(') || ch.equals('[') || ch.equals('{');
            if(condition){
                stack.push(ch);
            } else if(ch.equals(')')){
                if(stack.empty()) return false;
                else if(!stack.peek().equals('(')) return false;
                else stack.pop();
            } else if(ch.equals(']')){
                if(stack.empty()) return false;
                else if(!stack.peek().equals('[')) return false;
                else stack.pop();
            } else if(ch.equals('}')){
                if(stack.empty()) return false;
                else if(!stack.peek().equals('{')) return false;
                else stack.pop();
            } else {
                throw new IllegalArgumentException("Incorrect Character: '" + ch + "'");
            }
        }
        
        return stack.empty();
    }
}